Flaky test KubeConfigUtilsMergeTest.userInfoPreservedFromOriginalConfig

```
Error:  Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.283 s <<< FAILURE! -- in When merging multiple Configs with context argument, use context from ORIGINAL Config that matches context argument
Error:  io.fabric8.kubernetes.client.internal.KubeConfigUtilsMergeTest.userInfoPreservedFromOriginalConfig -- Time elapsed: 0.182 s <<< FAILURE!
java.lang.AssertionError: 

Expecting
  SundrioConfig(trustCerts=false, disableHostnameVerification=false, masterUrl=https://original-cluster.example.com/, apiVersion=v1, namespace=null, defaultNamespace=true, caCertFile=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/config-source-precedence/ca.crt, caCertData=null, clientCertFile=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/config-source-precedence/client.crt, clientCertData=null, clientKeyFile=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/config-source-precedence/client.key, clientKeyData=null, clientKeyAlgo=null, clientKeyPassphrase=changeit, trustStoreFile=null, trustStorePassphrase=null, keyStoreFile=null, keyStorePassphrase=null, authProvider=null, username=original-username, *** oauthToken=null, autoOAuthToken=token-from-kubeconfig, oauthTokenProvider=null, websocketPingInterval=30000, connectionTimeout=10000, maxConcurrentRequests=64, maxConcurrentRequestsPerHost=5, contexts=[NamedContext(context=Context(cluster=missing-entries, extensions=[], namespace=null, user=missing-entries, additionalProperties={}), name=missing-entries, additionalProperties={KUBERNETES_CONFIG_CONTEXT_FILE_KEY=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/internal/kube-config-utils-merge/config-4.yaml}), NamedContext(context=Context(cluster=original-cluster, extensions=[], namespace=null, user=original-user, additionalProperties={}), name=context-in-original-config, additionalProperties={}), NamedContext(context=Context(cluster=config-3-special-cluster, extensions=[], namespace=config-3-namespace, user=config-3-special-user, additionalProperties={}), name=context-in-config-3, additionalProperties={KUBERNETES_CONFIG_CONTEXT_FILE_KEY=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/internal/kube-config-utils-merge/config-3.yaml}), NamedContext(context=Context(cluster=config-1-cluster-2, extensions=[], namespace=null, user=config-1-user-2, additionalProperties={}), name=config-1-context-2, additionalProperties={KUBERNETES_CONFIG_CONTEXT_FILE_KEY=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/internal/kube-config-utils-merge/config-1.yaml}), NamedContext(context=Context(cluster=config-1-cluster, extensions=[], namespace=config-1-namespace, user=config-1-user, additionalProperties={}), name=context-in-all-configs, additionalProperties={KUBERNETES_CONFIG_CONTEXT_FILE_KEY=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/internal/kube-config-utils-merge/config-1.yaml}), NamedContext(context=Context(cluster=duplicate-cluster, extensions=[], namespace=null, user=duplicate-user, additionalProperties={}), name=duplicate-context, additionalProperties={KUBERNETES_CONFIG_CONTEXT_FILE_KEY=/home/runner/work/kubernetes-client/kubernetes-client/kubernetes-client-api/target/test-classes/internal/kube-config-utils-merge/config-1.yaml})], currentContext=NamedContext(context=Context(cluster=original-cluster, extensions=[], namespace=null, user=original-user, additionalProperties={}), name=context-in-original-config, additionalProperties={}), watchReconnectInterval=1000, watchReconnectLimit=-1, uploadRequestTimeout=120000, requestRetryBackoffLimit=10, requestRetryBackoffInterval=100, requestTimeout=10000, scaleTimeout=600000, loggingInterval=20000, impersonateUsername=null, impersonateGroups=[], impersonateExtras={}, http2Disable=false, httpProxy=null, httpsProxy=socks5://proxy.original-cluster.example.com, proxyUsername=null, proxyPassword=null, noProxy=null, userAgent=fabric8-kubernetes-client/999-SNAPSHOT, tlsVersions=[TLS_1_3, TLS_1_2], tlsServerName=null, onlyHttpWatches=false, watchList=null, customHeaders={}, autoConfigure=true, additionalProperties={})
to have a property or a field named "autoOAuthToken" with value
  null
but value was:
  "token-from-kubeconfig"
(static and synthetic fields are ignored)
	at io.fabric8.kubernetes.client.internal.KubeConfigUtilsMergeTest$ContextArgumentInOriginalConfig.userInfoPreservedFromOriginalConfig(KubeConfigUtilsMergeTest.java:336)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

--- a/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/ConfigSourcePrecedenceTest.java
+++ b/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/ConfigSourcePrecedenceTest.java
@@ -34,6 +34,11 @@ class ConfigSourcePrecedenceTest {
           Utils.filePath(ConfigSourcePrecedenceTest.class.getResource("/config-source-precedence/kube-config")));
     }
 
+    @AfterEach
+    void tearDown() {
+      System.clearProperty("kubeconfig");
+    }
+
     @Test
     @DisplayName("then use kubeconfig attributes in Config")
     void whenNoOtherSourceProvided_thenUseKubeConfig() {

--- a/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/internal/KubeConfigUtilsMergeTest.java
+++ b/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/internal/KubeConfigUtilsMergeTest.java
@@ -293,7 +293,7 @@ class KubeConfigUtilsMergeTest {
 
     @BeforeEach
     void setUp() {
-      result = new ConfigBuilder()
+      result = new ConfigBuilder(Config.empty())
           .addToContexts(new NamedContextBuilder()
               .withName("context-in-original-config")
               .withNewContext()

